users/views.py

```diff
@@ -164,4 +171,279 @@ def get(self, request, user_id, token):
+                email_response = requests.get(
+                    "https://api.github.com/user/emails", 
+                    headers=headers
+                )
+                email_response.raise_for_status()
+                emails = email_response.json()
+                primary_email = next(
+                    (e for e in emails if e.get("primary")), 
+                    None
+                )
+                if primary_email:
+                    email = primary_email.get("email")
+
+            if not email:
+                return Response(
+                    {"error": "Email not provided by GitHub"},
+                    status=status.HTTP_400_BAD_REQUEST
+                )
+                
+            # Process name
+            username = user_data.get("login")
+            name_parts = user_data.get("name", "").split(" ", 1)
+            first_name = name_parts[0] if name_parts else ""
+            last_name = name_parts[1] if len(name_parts) > 1 else ""
+
+            user, _ = self.get_or_create_user(
+                email=email,
+                defaults={
+                    # "username": username,
+                    "first_name": first_name,
+                    "last_name": last_name,
+                    "password": "",
+                    "user_phone": "",
+                    "title": "",
+                    "role": get_default_role()
+                }
+            )
+
+            return self.generate_jwt_response(user)
+            
+        except requests.RequestException:
+            return Response(
+                {"error": "Invalid GitHub token"}, 
+                status=status.HTTP_400_BAD_REQUEST
+            )
+    
+    def get_or_create_user(self, email, defaults):
+        """
+        Helper method to get or create user with intelligent field updates.
+        
+        Args:
+            email: User's email address (primary lookup key)
+            defaults: Dictionary of default values for new user creation
+            
+        Returns:
+            tuple: (user_object, created_bool) where:
+                - user_object: The retrieved or created user instance
+                - created_bool: Boolean indicating if user was created
+        """
+        user, created = User.objects.get_or_create(
+            email=email,
+            defaults=defaults
+        )
+        
+        if created:
+            user.set_unusable_password()
+            user.save()
+        else:
+            update_fields = {}
+            if not user.first_name and defaults.get('first_name'):
+                update_fields['first_name'] = defaults['first_name']
+            if not user.last_name and defaults.get('last_name'):
+                update_fields['last_name'] = defaults['last_name']
+            
+            if update_fields:
+                User.objects.filter(pk=user.pk).update(**update_fields)
+                user.refresh_from_db()
+        
+        return user, created
```

Comment:
Document what happens if a user already exists with a different role.
Add logging for user creation and updates for traceability.